### PR TITLE
Minor fixes for mlc and a CI test + Improvements for a package's initialisation

### PR DESCRIPTION
**Pull request by nmnobre**

### CODE REVIEW

**Review by arporter — CHANGES_REQUESTED** [review]
Thanks for fixing these things Nuno.
All tests and examples are OK with compilation.
Only suggestion I have is to try removing the `__all__` list from the `__init__.py` as I suspect we don't need it.

File: src/psyclone/psyir/transformations/__init__.py [hunk 1]
```diff
@@ -63,40 +63,43 @@
...
+from psyclone.psyir.transformations.read_only_verify_trans \
+    import ReadOnlyVerifyTrans
+from psyclone.psyir.transformations.region_trans import RegionTrans
 
 # The entities in the __all__ list are made available to import directly from
 # this package e.g.:
 # from psyclone.psyir.transformations import ExtractTrans
 
-__all__ = ['ChunkLoopTrans',
+__all__ = ['TransformationError',
```
[the remark this answers is not in the document]
Comment (nmnobre) [reply src/psyclone/psyir/transformations/__init__.py]: So I take it you agree we can't do this as it stands?

File: src/psyclone/psyir/transformations/__init__.py [hunk 2]
```diff
@@ -40,21 +40,21 @@
 transformations and base classes.
 '''
 
+from psyclone.psyir.transformations.transformation_error \
```
[the remark this answers is not in the document]
Comment (nmnobre) [reply src/psyclone/psyir/transformations/__init__.py]: Done, I just needed to adjust two other files :)
Apologies for the force pushing... I was careful not to rewrite the history prior to your review/comments though.

**Review by arporter — APPROVED** [review]
All requested changes have been made.
I've taken the liberty of fixing the one (very minor) conflict with master.
Tests and examples are all OK with compilation.
Ref guide builds with no new warnings.

**Pull request merged**